### DISCUSSION TITLE
"ERROR: Minions returned with non-zero exit" code - when running "sudo salt \* state.highstate"

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#7922] [Unsupported Versions]
When attempting to run "sudo salt \* state.highstate" to fully apply the disabling of SIDs I'm getting the following returned:

sensor-name-1_sensor:
    Data failed to compile:
    The function "state.highstate" is running as PID 2347 and was started at  with jid req

sensor-name-2_sensor:
    Data failed to compile:
    The function "state.highstate" is running as PID 12964 and was started at  with jid req

sensor-name-3_searchnode:
    Data failed to compile:
    The function "state.highstate" is running as PID 23569 and was started at  with jid req

sensor-name-4_searchnode:
    Data failed to compile:
    The function "state.highstate" is running as PID 6579 and was started at  with jid req

In an attempt to troubleshoot this, I have verified that the sensors returned from the previous cmd are up (via sudo salt-run manage.status). I have also ran the "sudo salt-run manage.down removekeys=True" cmd in an attempt to suppress any bad keys. I have also ran "sudo systemctl restart salt-minion" but that has not helped.

Apologies if this a stupid question as I'm relatively inexperienced with Linux, but could I just kill the PIDs returned from the "sudo salt \* state.highstate" cmd and try to re-run the cmd? If anybody has any suggestions or has encountered a similar problem your help will be much appreciated. Thanks!

**Comment by dougburks** [comment]
Usually if highstate is already running, you can simply wait a minute or two for it to complete.

**Reply by ghost** [reply]
Thanks for your response, Doug!

I've tried this a few times and it just appears as though state.highstate is continually running on the sensors (the PIDs are changing). Again, the error of "ERROR: Minions returned with non-zero exit code" is being returned.

**Reply by dougburks** [reply]
The root of this question was about disabling SIDs and if the highstates are completing properly then the SIDs should be disabled. Are you still receiving alerts for the disabled SIDs?

**Reply by ghost** [reply]
Yes, I'm still receiving alerts for the disabled SIDs.

**Reply by dougburks** [reply]
Have you checked `/opt/so/log/salt/` on the manager and minions for clues?

You could try killing the process and running it manually to examine the output for additional clues.
